src/plugins/ResizeReorder/ResizerKnob.js

```diff
@@ -0,0 +1,255 @@
+      this.props.touchEnabled,
+    );
+    this.mouseMoveTracker.captureMouseMoves(event);
+  };
+
+  /**
+   * @param {MouseEvent} ev Mouse down event
+   */
+  onMouseDown = (ev) => {
+    this.initializeDOMMouseMoveTracker(ev);
+    const initialMouseXCoordinate = FixedDataTableEventHelper.getCoordinatesFromEvent(ev).x;
+    this.setState({
+      initialMouseXCoordinate,
+      isColumnResizing: true,
+      totalDisplacement: 0,
+      currentMouseXCoordinate: initialMouseXCoordinate,
+    });
+  };
+
+  onMouseUp = () => {
+    const { minWidth, maxWidth } = this.getMinMaxWidth();
+    const newWidth = clamp(
+      this.props.width +
+      this.state.totalDisplacement * (this.props.isRTL ? -1 : 1),
+      minWidth,
+      maxWidth,
+    );
+    this.mouseMoveTracker.releaseMouseMoves();
+    this.resetColumnResizing(() =>
+      this.props.onColumnResizeEnd(newWidth, this.props.columnKey)
+    );
+  };
+
+  /**
+   *
+   * @param {number} displacementX Displacement of mouse along x-direction
+   */
+  onMouseMove = (displacementX) => {
+    const {
+      initialMouseXCoordinate,
+      totalDisplacement: previousTotalDisplacement,
+    } = this.state;
+    // displacementX is negative if movement is in left direction
+    const newTotalDisplacement = previousTotalDisplacement + displacementX;
+    let newResizerLineXCoordinate =
+      initialMouseXCoordinate + newTotalDisplacement;
+    const { minWidth, maxWidth } = this.getMinMaxWidth();
+    // Limit the resizer line to not move ahead or back of maxWidth and minWidth respectively
+    if (
+      this.props.width + newTotalDisplacement < minWidth ||
+      this.props.width + newTotalDisplacement > maxWidth
+    ) {
+      // If new position is going out of bounds, instead of updating,  use the previous value
+      newResizerLineXCoordinate = this.state.currentMouseXCoordinate;
+    }
+    this.setState({
+      totalDisplacement: newTotalDisplacement,
+      currentMouseXCoordinate: newResizerLineXCoordinate,
+    });
+  };
+
+  /**
+   * Set isColumnResizing to false to hide the ResizerLine and set displacement to 0
+   * @param {Function} callback Called after resetting state
+   */
+  resetColumnResizing = (callback) => {
+    this.setState(
+      {
+        isColumnResizing: false,
+        totalDisplacement: 0,
+      },
+      () => {
+        if (_.isFunction(callback)) {
+          callback();
+        }
+      },
+    );
+  };
+
+  /**
+   * If minWidth and Width not given, mapping them to a valid range i.e. 0 to Number.MAX_SAFE_INTEGER
+   * @returns {{minWidth: number, maxWidth: number}}
+   */
+  getMinMaxWidth = () => {
+    return {
+      minWidth: this.props.minWidth || 0,
+      maxWidth: this.props.maxWidth || Number.MAX_SAFE_INTEGER,
+    };
+  };
+
+  /**
+   * @param {Object} event
+   */
+  suppressEvent = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+  };
+}
+
+export default ResizerKnob;
+
+ResizerKnob.propTypes = {
```

Comment:
here and elsewhere: Are the required props marked as `required`?

Reply:
Done